x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/TrainedModelAssignmentRebalancer.java

```diff
@@ -81,7 +86,58 @@ TrainedModelAssignmentMetadata.Builder rebalance() {
         }
 
         AssignmentPlan assignmentPlan = computeAssignmentPlan();
-        return buildAssignmentsFromPlan(assignmentPlan);
+        TrainedModelAssignmentMetadata.Builder builder = buildAssignmentsFromPlan(assignmentPlan);
+
+        if (shuttingDownNodeIds.isEmpty()) {
+            return builder;
+        }
+
+        return setShuttingDownNodeRoutesToStopping(builder);
+    }
+
+    private TrainedModelAssignmentMetadata.Builder setShuttingDownNodeRoutesToStopping(TrainedModelAssignmentMetadata.Builder builder) {
+        for (TrainedModelAssignment existingAssignment : currentMetadata.allAssignments().values()) {
+            boolean foundShuttingDownNodeForAssignment = false;
+
+            String existingDeploymentId = existingAssignment.getDeploymentId();
+            // TODO this may be overkill, based on my limited testing, the builder will always have the assignment so maybe that's
+            // guaranteed and we don't need to check?
+            TrainedModelAssignment.Builder assignmentBuilder = builder.hasModelDeployment(existingAssignment.getDeploymentId())
```

Comment:
Based on my limited testing, the passed in builder will always have the model deployment so maybe it's not worth checking? 🤷‍♂️